x/fungible/keeper/zeta.go

```diff
@@ -9,9 +10,24 @@
 	"github.com/zeta-chain/node/x/fungible/types"
 )
 
+// ZETAMaxSupplyStr is the maximum mintable ZETA in the fungible module
+// 1.85 billion ZETA
+const ZETAMaxSupplyStr = "1850000000000000000000000000"
+
 // MintZetaToEVMAccount mints ZETA (gas token) to the given address
 // NOTE: this method should be used with a temporary context, and it should not be committed if the method returns an error
 func (k *Keeper) MintZetaToEVMAccount(ctx sdk.Context, to sdk.AccAddress, amount *big.Int) error {
+	zetaMaxSupply, ok := sdk.NewIntFromString(ZETAMaxSupplyStr)
+	if !ok {
+		return errors.New("failed to parse ZETA max supply")
+	}
+
+	// Check if the max supply is reached
+	supply := k.bankKeeper.GetSupply(ctx, config.BaseDenom)
+	if supply.Amount.Add(sdk.NewIntFromBigInt(amount)).GT(zetaMaxSupply) {
+		return types.ErrMaxSupplyReached
+	}
```

Comment:
agree

Reply:
I will apply for develop here for code cleanliness but I think it's fine the parsing error is more a developer error handling since we handle a constant